### DISCUSSION TITLE
Debugging navigationTimeout errors

### DISCUSSION THREAD

**Discussion by spinningarrow** [getgauge/taiko#1802] [Development]
We have some recent issues where a navigationTimeout error occurs after a `click` or a `goto`. The actual navigation happens successfully well before the timeout limit. Is there a way to debug why Taiko is unable to detect the navigation? `.trace` doesn't really provide enough insight into what Taiko is looking for.

**Comment by spinningarrow** [comment]
Update: I tried [enabling Taiko's debug log events](https://docs.taiko.dev/frequently_asked_questions/#how-do-i-enable-debug-log-events%3F) and got wayy too much output. Are there specific events I can filter for related to why navigation is not detected?

**Comment by NivedhaSenthil** [comment]
Could be because Taiko is waiting for some network request which is not resolved, from logs you could see if there is any requestID without a matching resolved event.
yes currently it is tricky getting around the logs need to find a better way.

If it is enough to wait for the page load in your case, you could try passing `waitForEvents:['loadEventFired']` as an option to `goto` or `click` so that Taiko wont wait for any request triggered along with the action.

**Comment by spinningarrow** [comment]
Thanks Nivedha! For now the problem magically solved itself by upgrading to
the latest Taiko. If it comes back I’ll look into it more with your
suggestions.

**Comment by smv-orrin** [comment]
we have seen a similar issue that has to do with the default action of the click() api.  Basically, click() api calls default to use waitForNavigation: true, which seems to be starting a countdown to a navigation event.  When there is no navigation (because it's just an on-page thing, like clicking a button or other non-navigating element) it just keeps waiting until it times out, unless you're lucky and your test beats the default navigationTimeout.  We got around this by passing {waitForNavigation: false} during non-navigating click() api calls.
